x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_threat_matching.ts

```diff
@@ -383,40 +395,37 @@ export default ({ getService }: FtrProviderContext) => {
           expect(signalsOpen.hits.hits.length).equal(1);
 
           const { hits } = signalsOpen.hits;
-          const threats = hits.map((hit) => hit._source.threat);
-          expect(threats).to.eql([
+          const [threat] = hits.map((hit) => hit._source.threat) as Array<{ indicator: unknown[] }>;
+
+          assertContains(threat.indicator, [
             {
-              indicator: [
-                {
-                  description: 'this should match auditbeat/hosts on both port and ip',
-                  first_seen: '2021-01-26T11:06:03.000Z',
-                  ip: '45.115.45.3',
-                  matched: {
-                    atomic: '45.115.45.3',
-                    id: '978785',
-                    index: 'filebeat-8.0.0-2021.01.26-000001',
-                    field: 'source.ip',
-                    type: 'url',
-                  },
-                  port: 57324,
-                  provider: 'geenensp',
-                  type: 'url',
-                },
-                {
-                  description: 'this should match auditbeat/hosts on ip',
-                  first_seen: '2021-01-26T11:06:03.000Z',
-                  ip: '45.115.45.3',
-                  matched: {
-                    atomic: '45.115.45.3',
-                    id: '978787',
-                    index: 'filebeat-8.0.0-2021.01.26-000001',
-                    field: 'source.ip',
-                    type: 'ip',
-                  },
-                  provider: 'other_provider',
-                  type: 'ip',
-                },
-              ],
+              description: 'this should match auditbeat/hosts on both port and ip',
+              first_seen: '2021-01-26T11:06:03.000Z',
+              ip: '45.115.45.3',
+              matched: {
+                atomic: '45.115.45.3',
+                id: '978785',
+                index: 'filebeat-8.0.0-2021.01.26-000001',
+                field: 'source.ip',
+                type: 'url',
+              },
+              port: 57324,
+              provider: 'geenensp',
+              type: 'url',
+            },
+            {
+              description: 'this should match auditbeat/hosts on ip',
+              first_seen: '2021-01-26T11:06:03.000Z',
+              ip: '45.115.45.3',
+              matched: {
+                atomic: '45.115.45.3',
+                id: '978787',
+                index: 'filebeat-8.0.0-2021.01.26-000001',
+                field: 'source.ip',
+                type: 'ip',
+              },
+              provider: 'other_provider',
+              type: 'ip',
```

Comment:
If I got it right, these tests load some events from `auditbeat/hosts` ES archive, create a rule, generate signals and then check that signals have certain fields (set by the enrichment process in the rule?) like `matched.*`.

I think my question is: how stable or brittle these tests are in terms of input data (events being tested)? How likely it is if someone updates this ES archive with new events, and these tests will break with false negatives although the enrichment functionality stays correctly implemented?

Reply:
Good question! My understanding is that those archives are stable, and that for e.g. auditbeat-7.13 one would create a new archive/tests rather than update the existing one.

The tests are definitely brittle in terms of these data changing, but I believe that most if not all of our event-based integration tests share this property. Most of the other "create rule, assert on alerts" tests like this are reliant on the same archives, it just happened that these tests were brittle due to the nondeterministic ordering of named queries.

Reply:
> My understanding is that those archives are stable, and that for e.g. auditbeat-7.13 one would create a new archive/tests rather than update the existing one.

Oh sorry, I was assuming they could be updated from time to time. :) Good to know that it's not the case. Thanks for your explanation @rylnd 👍